BackendApartmentReservation/BackendApartmentReservation/Repositories/TripRepository.cs

```diff
@@ -34,5 +36,43 @@ public async Task<IEnumerable<DbEmployeeAmenitiesChecklist>> GetTripChecklistsWi
                 .Include(c => c.LivingPlace)
                 .ToListAsync();
         }
+
+        public async Task<DbTrip> CreateTrip (CreateTripRequest tripRequest)
+        {
+            if (tripRequest.DepartureDate != null && tripRequest.ReturnDate != null)
+            {
+                var destinationOffice = await _db.Offices.FirstOrDefaultAsync(o => o.Id == tripRequest.DestinationOfficeId);
+                var employees = await _db.Employees
+                    .Include(e => e.Office)
+                    .Where(e => tripRequest.UserIds.Contains(e.Id))
+                    .ToListAsync();
+
+                var groups = (from e in employees
+                             group e by e.Office into g

```

Comment:
For safety measure, group by office id instead, I am not sure how the comparison will be done in this case